Tests/Integration/Core/VaultTest.php

```diff
@@ -0,0 +1,210 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+use Wirecard\Oxid\Core\OxidEeEvents;
+use Wirecard\Oxid\Core\Vault;
+use Wirecard\PaymentSdk\Response\SuccessResponse;
+
+class VaultTest extends \Wirecard\Test\WdUnitTestCase
+{
+    protected function dbData()
+    {
+        $oFutureDate = new DateTime();
+        $oFutureDate->add(new DateInterval('P1Y'));
+
+        $oPastDate = new DateTime();
+        $oPastDate->sub(new DateInterval('P1Y'));
+
+        return [
+            [
+                'table' => OxidEeEvents::VAULT_TABLE,
+                'columns' => ['OXID', 'USERID', 'ADDRESSID', 'TOKEN', 'MASKEDPAN', 'EXPIRATIONMONTH', 'EXPIRATIONYEAR'],
+                'rows' => [
+                    [1, 'User ID 1', 'Address ID 1', 'Token 1', 'Masked****Pan', 2, (int) $oFutureDate->format('y')],
+                    [2, 'User ID 1', 'Address ID 1', 'Token 2', 'Masked****Pan', 2, (int) $oPastDate->format('y')],
+                    [3, 'User ID 2', 'Address ID 2', 'Token 3', 'Masked****Pan', 2, (int) $oFutureDate->format('y')],
+                    [4, 'User ID 2', 'Address ID 2', 'Token 4', 'Masked****Pan', 3, (int) $oFutureDate->format('y')],
+                    [5, 'User ID 3', 'Address ID 3', 'Token 5', 'Masked****Pan', 2, (int) $oPastDate->format('y')],
+                    [6, 'User ID 4', 'Address ID 4', 'Token 6', 'Masked****Pan', 2, (int) $oFutureDate->format('y')],
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider saveCardProvider
+     */
+    public function testSaveCard($sToken, $sAddressId, $sUserId, $aExpected)
+    {
+        $oSuccessResponse = $this->getMockBuilder(SuccessResponse::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $oSuccessResponse->method('getCardTokenId')
+            ->willReturn($sToken);
+        $oSuccessResponse->method('getMaskedAccountNumber')
+            ->willReturn('Masked Account Number');
+
+        $aCard = [
+            'expiration-month' => 9,
+            'expiration-year' => 21,
+        ];
+
+        $oUser = $this->getMockBuilder(\OxidEsales\EshopCommunity\Application\Model\User::class)
```

Comment:
This would be more readable if the dependency was imported at the top via `use` (applies to the whole file).

Reply:
+1

Reply:
👍